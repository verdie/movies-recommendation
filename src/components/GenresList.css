.wrap{
  width: 1024px;
  margin: 0 auto;
  -webkit-box-shadow: 0px 0px 16px 2px rgba(189,171,136,0.61);
  -moz-box-shadow: 0px 0px 16px 2px rgba(189,171,136,0.61);
  box-shadow: 0px 0px 16px 2px rgba(189,171,136,0.61);
  background-color: rgb(254, 250, 245);
  padding-bottom: 30px;
}

.category-wrapper{
  display: flex;
  flex-wrap: wrap;
  padding: 20px 20px;
}
h2{
  text-align: center;
  padding-top: 20px;
  color:  rgb(130, 111, 89)
}
a {
  text-decoration: none;
  color:  rgb(130, 111, 89);
  font-size: 20px;
  width: 22%;
  margin: 0 14px;
 
}
.genre-p:hover{
  background-color: rgb(247, 242, 235);
    box-shadow: 0 5px 20px rgb(227, 207, 184);
    transform: translateY(-3px);
}

.genre-p{
  padding: 20px 20px;
  text-align: center;
  font-weight: bolder;
  border: 3px solid rgb(227, 207, 184);
  box-shadow: 0 3px 20px  rgb(182, 160, 135);
}

.genre-p:active{
  background-color: rgb(247, 242, 235);
}