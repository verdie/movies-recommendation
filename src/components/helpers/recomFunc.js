export default function recomFunc(values) {
  var counts = {};
  values
    .flat()
    .forEach((movie) => {
      if (!counts[movie.id]) 
        counts[movie.id] = {
          times: 0
        };
      counts[movie.id].times = counts[movie.id].times + 1;
      counts[movie.id].movie = movie;
    });

  return Object
    .values(counts)
    .sort((a, b) => b.times - a.times);
}
