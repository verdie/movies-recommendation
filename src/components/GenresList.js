import React from 'react';
import * as request from 'superagent'
import {Link} from 'react-router-dom'
import './GenresList.css'

class GenresList extends React.Component {
  constructor() {
    super()
    this.state = {
      gens: []
    };
  }
  componentDidMount() {
    request
      .get('https://api.themoviedb.org/3/genre/movie/list?api_key=9f331ae5aeb7c6a0d55d497451' +
        'cb6fca&language=en-US')
      .then(response => {
        this.setState({gens: response.body.genres})
      })
  }
  render() {
    console.log('STATE', this.state)
    return (
      <div className='wrap'>
        <h2>Please select category that you like</h2>
        <div className='category-wrapper'>
          {this.state.gens.length
            ? this.state.gens.map(item => {
                return (
                  <Link key={item.id} to={`/genre/${item.id}`}>
                    <p className='genre-p'>{item.name}</p>
                  </Link>
                );
              })
            : null
          }
        </div>
      </div>
    )
  }
}

export default GenresList;
