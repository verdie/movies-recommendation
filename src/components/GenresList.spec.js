import React from 'react'
import { shallow } from 'enzyme'
import GenresList from './GenresList'
import toJson from 'enzyme-to-json'
import recomFunc from './helpers/recomFunc.js';
const mockGenres = [{"id":28,"name":"Action"},{"id":12,"name":"Adventure"},{"id":16,"name":"Animation"},{"id":35,"name":"Comedy"},{"id":80,"name":"Crime"},{"id":99,"name":"Documentary"},{"id":18,"name":"Drama"},{"id":10751,"name":"Family"},{"id":14,"name":"Fantasy"},{"id":36,"name":"History"},{"id":27,"name":"Horror"},{"id":10402,"name":"Music"},{"id":9648,"name":"Mystery"},{"id":10749,"name":"Romance"},{"id":878,"name":"Science Fiction"},{"id":10770,"name":"TV Movie"},{"id":53,"name":"Thriller"},{"id":10752,"name":"War"},{"id":37,"name":"Western"}];

describe('GenresList component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GenresList />)
    wrapper.setState({
      gens: mockGenres
    });

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders correctly', () => {
    const mock = [[{id: 1}, {id: 2}], [{id: 2}, {id: 3}]]
    const result = recomFunc(mock);
    expect(result[0].times).toBe(2)
    expect(result[0].movie.id).toBe(2)
    expect(result[1].times).toBe(1)
    expect(result[1].movie.id).toBe(1)
    expect(result[2].times).toBe(1)
    expect(result[2].movie.id).toBe(3)
  })
})
