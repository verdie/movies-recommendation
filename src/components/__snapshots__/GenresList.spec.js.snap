// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenresList component renders correctly 1`] = `
<div
  className="wrap"
>
  <h2>
    Please select category that you like
  </h2>
  <div
    className="category-wrapper"
  >
    <Link
      key="28"
      to="/genre/28"
    >
      <p
        className="genre-p"
      >
        Action
      </p>
    </Link>
    <Link
      key="12"
      to="/genre/12"
    >
      <p
        className="genre-p"
      >
        Adventure
      </p>
    </Link>
    <Link
      key="16"
      to="/genre/16"
    >
      <p
        className="genre-p"
      >
        Animation
      </p>
    </Link>
    <Link
      key="35"
      to="/genre/35"
    >
      <p
        className="genre-p"
      >
        Comedy
      </p>
    </Link>
    <Link
      key="80"
      to="/genre/80"
    >
      <p
        className="genre-p"
      >
        Crime
      </p>
    </Link>
    <Link
      key="99"
      to="/genre/99"
    >
      <p
        className="genre-p"
      >
        Documentary
      </p>
    </Link>
    <Link
      key="18"
      to="/genre/18"
    >
      <p
        className="genre-p"
      >
        Drama
      </p>
    </Link>
    <Link
      key="10751"
      to="/genre/10751"
    >
      <p
        className="genre-p"
      >
        Family
      </p>
    </Link>
    <Link
      key="14"
      to="/genre/14"
    >
      <p
        className="genre-p"
      >
        Fantasy
      </p>
    </Link>
    <Link
      key="36"
      to="/genre/36"
    >
      <p
        className="genre-p"
      >
        History
      </p>
    </Link>
    <Link
      key="27"
      to="/genre/27"
    >
      <p
        className="genre-p"
      >
        Horror
      </p>
    </Link>
    <Link
      key="10402"
      to="/genre/10402"
    >
      <p
        className="genre-p"
      >
        Music
      </p>
    </Link>
    <Link
      key="9648"
      to="/genre/9648"
    >
      <p
        className="genre-p"
      >
        Mystery
      </p>
    </Link>
    <Link
      key="10749"
      to="/genre/10749"
    >
      <p
        className="genre-p"
      >
        Romance
      </p>
    </Link>
    <Link
      key="878"
      to="/genre/878"
    >
      <p
        className="genre-p"
      >
        Science Fiction
      </p>
    </Link>
    <Link
      key="10770"
      to="/genre/10770"
    >
      <p
        className="genre-p"
      >
        TV Movie
      </p>
    </Link>
    <Link
      key="53"
      to="/genre/53"
    >
      <p
        className="genre-p"
      >
        Thriller
      </p>
    </Link>
    <Link
      key="10752"
      to="/genre/10752"
    >
      <p
        className="genre-p"
      >
        War
      </p>
    </Link>
    <Link
      key="37"
      to="/genre/37"
    >
      <p
        className="genre-p"
      >
        Western
      </p>
    </Link>
  </div>
</div>
`;
